eureka2-client/src/test/java/com/netflix/eureka2/client/interest/BatchingRegistryImplTest.java

```diff
@@ -0,0 +1,87 @@
+package com.netflix.eureka2.client.interest;
+
+import com.netflix.eureka2.interests.ChangeNotification;
+import com.netflix.eureka2.interests.Interest;
+import com.netflix.eureka2.interests.Interests;
+import com.netflix.eureka2.interests.StreamStateNotification;
+import com.netflix.eureka2.interests.StreamStateNotification.BufferingState;
+import com.netflix.eureka2.registry.instance.InstanceInfo;
+import com.netflix.eureka2.rx.ExtTestSubscriber;
+import org.junit.Before;
+import org.junit.Test;
+import rx.Observable;
+import rx.subjects.PublishSubject;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.assertThat;
+
+/**
+ * @author Tomasz Bak
+ */
+public class BatchingRegistryImplTest {
```

Comment:
I think we are missing tests for the following cases:
- onError and onCompleted cases for the attached data stream
- behaviours for reconnecting data streams (this will be a common occurrence)
- case tests for the many different cases of shouldBatch
- shutdown cleanup test